Sim_vehicle.py fail

~/ardupilot/ArduCopter$ sim_vehicle.py --console --map
SIM_VEHICLE: Start
SIM_VEHICLE: Killing tasks
SIM_VEHICLE: Starting up at -35.363261,149.165230,584,353 (CMAC)
SIM_VEHICLE: WAF build
SIM_VEHICLE: Configure waf
SIM_VEHICLE: "/home/user/ardupilot/modules/waf/waf-light" "configure" "--board" "sitl"
Setting top to                           : /home/user/ardupilot 
Setting out to                           : /home/user/ardupilot/build 
Autoconfiguration                        : enabled 
Setting board to                         : sitl 
Using toolchain                          : native 
Checking for 'g++' (C++ compiler)        : /usr/lib/ccache/g++ 
Checking for 'gcc' (C compiler)          : /usr/lib/ccache/gcc 
Checking for c flags '-MMD'              : yes 
Checking for cxx flags '-MMD'            : yes 
Checking for need to link with librt     : not necessary 
Checking for HAVE_CMATH_ISFINITE         : yes 
Checking for HAVE_CMATH_ISINF            : yes 
Checking for HAVE_CMATH_ISNAN            : yes 
Checking for NEED_CMATH_ISFINITE_STD_NAMESPACE : yes 
Checking for NEED_CMATH_ISINF_STD_NAMESPACE    : yes 
Checking for NEED_CMATH_ISNAN_STD_NAMESPACE    : yes 
Checking for header endian.h                   : yes 
Checking for header byteswap.h                 : yes 
Checking for HAVE_MEMRCHR                      : yes 
Checking for program 'python'                  : /usr/bin/python 
Checking for python version >= 2.7.0           : 2.7.12 
Checking for program 'python'                  : /usr/bin/python 
Checking for python version >= 2.7.0           : 2.7.12 
Source is git repository                       : yes 
Update submodules                              : yes 
Checking for program 'git'                     : /usr/bin/git 
Checking for program 'size'                    : /usr/bin/size 
Benchmarks                                     : disabled 
Unit tests                                     : enabled 
Scripting                                      : disabled 
Scripting runtime checks                       : enabled 
Checking for program 'rsync'                   : /usr/bin/rsync 
'configure' finished successfully (0.908s)
SIM_VEHICLE: Building
SIM_VEHICLE: "/home/user/ardupilot/modules/waf/waf-light" "build" "--target" "bin/arducopter"
Waf: Entering directory `/home/user/ardupilot/build/sitl'
[3/4] Processing modules/mavlink/message_definitions/v1.0/ardupilotmega.xml
[4/4] Creating build/sitl/ap_version.h
.
.
.
[390/532] Compiling libraries/AP_HAL_SITL/UARTDriver.cpp
[391/532] Compiling libraries/AP_HAL_SITL/Storage.cpp
[392/532] Compiling libraries/AP_HAL_SITL/HAL_SITL_Class.cpp
../../libraries/AP_HAL_SITL/Storage.cpp: In member function ‘virtual void HALSITL::Storage::_timer_tick()’:
../../libraries/AP_HAL_SITL/Storage.cpp:125:44: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
         if (lseek(log_fd, offset, SEEK_SET) != offset) {
                                            ^
compilation terminated due to -Wfatal-errors.
cc1plus: some warnings being treated as errors

Waf: Leaving directory `/home/user/ardupilot/build/sitl'
Build failed
 -> task in 'objs/AP_HAL_SITL' failed (exit status 1): 
	{task 3040399060L: cxx Storage.cpp -> Storage.cpp.0.o}
 (run with -v to display more information)
SIM_VEHICLE: Build failed
SIM_VEHICLE: Killing tasks


16.04 ubuntu
_What version was the issue encountered with_

**Platform**
[  ] All
[  ] AntennaTracker
[x] Copter
[  ] Plane
[  ] Rover
[  ] Submarine

--- a/libraries/AP_HAL_SITL/Storage.cpp
+++ b/libraries/AP_HAL_SITL/Storage.cpp
@@ -121,7 +121,7 @@ void Storage::_timer_tick(void)
 
 #if STORAGE_USE_POSIX
     if (using_filesystem && log_fd != -1) {
-        uint32_t offset = STORAGE_LINE_SIZE*i;
+        const off_t offset = STORAGE_LINE_SIZE*i;
         if (lseek(log_fd, offset, SEEK_SET) != offset) {
             return;
         }

